packages/now-cli/src/commands/deploy/latest.js

```diff
@@ -260,6 +257,108 @@ export default async function main(
     );
   }
 
+  const client = new Client({
+    apiUrl: ctx.apiUrl,
+    token: ctx.authConfig.token,
+    debug: debugEnabled,
+  });
+
+  // retrieve `project` and `org` from .now
+  const link = await getLinkedProject(output, client, path);
+
+  if (link.status === 'error') {
+    return link.exitCode;
+  }
+
+  let { org, project, status } = link;
+  let newProjectName = null;
+  let rootDirectory = project ? project.rootDirectory : null;
+
+  if (status === 'not_linked') {
+    const shouldStartSetup =
+      autoConfirm ||
+      (await confirm(
+        `Set up and deploy ${chalk.cyan(`“${toHumanPath(path)}”`)}?`,
+        true
+      ));
+
+    if (!shouldStartSetup) {
+      output.print(`Aborted. Project not set up.\n`);
+      return 0;
+    }
+
+    org = await selectOrg(
+      output,
+      'Which scope do you want to deploy to?',
+      client,
+      ctx.config.currentTeam,
+      autoConfirm
+    );
+
+    // We use `localConfig` here to read the name
+    // even though the `now.json` file can change
+    // afterwards, this is fine since the property
+    // will be deprecated and can be replaced with
+    // user input.
+    const detectedProjectName = getProjectName({
+      argv,
+      nowConfig: localConfig,
+      isFile,
+      paths,
+    });
+
+    const projectOrNewProjectName = await inputProject(
+      output,
+      client,
+      org,
+      detectedProjectName,
+      autoConfirm
+    );
+
+    if (typeof projectOrNewProjectName === 'string') {
+      newProjectName = projectOrNewProjectName;
+      rootDirectory = await inputRootDirectory(path, output, autoConfirm);
+    } else {
+      project = projectOrNewProjectName;
+      rootDirectory = project.rootDirectory;
+
+      // we can already link the project
+      await linkFolderToProject(
+        output,
+        path,
+        {
+          projectId: project.id,
+          orgId: org.id,
+        },
+        project.name,
+        org.slug
+      );
+      status = 'linked';
+    }
+  }
+
+  const sourcePath = rootDirectory ? join(path, rootDirectory) : path;
+
+  if (
+    rootDirectory &&
+    (await validateRootDirectory(
+      output,
+      path,
+      sourcePath,
+      project
+        ? `To change your project settings, go to https://zeit.co/${org.slug}/${project.name}/settings`
+        : ''
+    )) === false
+  ) {
+    return 1;
+  }
+
+  // When there is no `now.json` file we'll try
+  // to get it from the root directory.
```

Comment:
I think we should we first look in the root directory, and then fallback to localConfig 🤔

Reply:
I don't think that would make much of a difference though. There should only be one `now.json` anyways. Or can you think of something else?

Reply:
I've also added a warning when the `now.json` file is outside of the root directory, because there should just be one way for this, and since `functions` have to use a relative path from the root directory it wouldn't make sense to put it elsewhere, therefore I added a warning.

`now dev` already supports this.

Reply:
Does `now dev` has a fallback to the root @AndyBitz ?

Reply:
No